containerd-shim-v2/service.go

```diff
@@ -72,13 +73,33 @@ func New(ctx context.Context, id string, publisher events.Publisher) (cdshim.Shi
 	vci.SetLogger(ctx, logger)
 	katautils.SetLogger(ctx, logger, logger.Logger.Level)
 
+	// load runtime config so that tracing can start if enabled
+	_, runtimeConfig, err := katautils.LoadConfiguration("", false, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// create tracer
+	tracer := opentracing.Tracer(nil)
```

Comment:
Why are you instantiating a tracer here?

Reply:
@amshinde New is the first function called in service.go, it must happen after the runtime config is loaded (also added to New in this PR), and it must be started after the logging is set or you get errors running a container.

Reply:
I meant this could be changed to a variable declaration  `var tracer opentracing.Tracer` as thats the approach we follow.
I also feel that you need not check for runtimeConfig.Trace here, as looking at the CreateTracer function here:
https://github.com/kata-containers/runtime/blob/d937c067f223324ef2e428adc5e0f807d74c4855/pkg/katautils/tracing.go#L38

it uses a NOP trace implementation if tracing is disabled.
cc @jodh-intel to clarify why we took this approach

Reply:
@amshinde Oh I see; I misunderstood.

Please correct me if I'm misunderstanding your point about the runtime config, but I think we do need to check if we want a complete trace. If we don't check for `runtimeConfig.Trace` here, we won't be able to create the tracer and start tracing until `Create` and will miss spans for `New`, `StartShim`, and `New` again, because they will be treated as NOP. See [this comment](https://github.com/kata-containers/runtime/pull/2980#issuecomment-698466878). We can skip a few spans and start tracing later, but it is my understanding that the runtime config must be loaded (or at least runtimeConfig.Trace must be set) before tracing can begin.

Reply:
@amshinde I see (again), the actual check for tracing being enabled doesn't need to happen.